ros2param/ros2param/verb/dump.py

```diff
@@ -0,0 +1,146 @@
+            value = pvalue.string_value
+        elif pvalue.type == ParameterType.PARAMETER_BYTE_ARRAY:
+            value = pvalue.byte_array_value
+        elif pvalue.type == ParameterType.PARAMETER_BOOL_ARRAY:
+            value = pvalue.bool_array_value
+        elif pvalue.type == ParameterType.PARAMETER_INTEGER_ARRAY:
+            value = pvalue.integer_array_value
+        elif pvalue.type == ParameterType.PARAMETER_DOUBLE_ARRAY:
+            value = pvalue.double_array_value
+        elif pvalue.type == ParameterType.PARAMETER_STRING_ARRAY:
+            value = pvalue.string_array_value
+        elif pvalue.type == ParameterType.PARAMETER_NOT_SET:
+            value = None
+        else:
+            value = None
+
+        return value
+
+    def insert_dict(self, dict, key, value):
+        split = key.split(PARAMETER_SEPARATOR_STRING, 1)
+        if len(split) > 1:
+            if not split[0] in dict:
+                dict[split[0]] = {}
+            self.insert_dict(dict[split[0]], split[1], value)
+        else:
+            dict[key] = value
+
+    def main(self, *, args):  # noqa: D102
+
+        with NodeStrategy(args) as node:
+            node_names = get_node_names(
+                node=node, include_hidden_nodes=args.include_hidden_nodes)
+
+        absolute_node_name = get_absolute_node_name(args.node_name)
+        node_name = parse_node_name(absolute_node_name)
+        if absolute_node_name:
+            if absolute_node_name not in [n.full_name for n in node_names]:
+                return 'Node not found'
+
+        if not os.path.isdir(args.file_path):
+            return 'Invalid output directory'
+
+        with DirectNode(args) as node:
+            # create client
+            service_name = '{absolute_node_name}/list_parameters'.format_map(locals())
+            client = node.create_client(ListParameters, service_name)
+
+            client.wait_for_service()
+
+            if not client.service_is_ready():
+                return 'Something went wrong'
+
+            request = ListParameters.Request()
+            future = client.call_async(request)
+
+            # wait for response
+            rclpy.spin_until_future_complete(node, future)
+
+            yaml_output = {node_name.name: {'ros__parameters': {}}}
+
+            # retrieve values
+            if future.result() is not None:
+                response = future.result()
+                for param_name in sorted(response.result.names):
+                    pval = self.get_parameter_value(node, absolute_node_name, param_name)
+                    self.insert_dict(
+                        yaml_output[node_name.name]['ros__parameters'], param_name, pval)
+            else:
+                e = future.exception()
+                print(
+                    'Exception while calling service of node '
+                    "'{node_name.full_name}': {e}".format_map(locals()),
+                    file=sys.stderr)
```

Comment:
I think the CLI tools expect this sort of thing to be [re-raised as `RuntimeError`](https://github.com/ros2/ros2cli/blob/96c9ff272b694d9c0f20ab59b87f13cee6e42125/ros2cli/ros2cli/cli.py#L72-L73). That would jump over the file writing part too, which I think makes sense if the service call fails.

Reply:
Should I `raise` on [l.120](https://github.com/ros2/ros2cli/pull/285/files/5dbe3def617fc5a38a7343c3cad36dd2686d1c96#diff-e09ffdcdc4d45ac31ebc5ac3b5f60d5bR120) too then ??

Reply:
Initial comment done.

Reply:
Raising to some other places too : 60d5b8551f7eda838c8d42f484ef1c970b5b1911